core/services/keeper/upkeep_executer.go

```diff
@@ -277,12 +277,15 @@ func (ex *UpkeepExecuter) estimateGasPrice(upkeep UpkeepRegistration) (gasPrice
 		return nil, fee, errors.Wrap(err, "unable to construct performUpkeep data")
 	}
 
+	// Context used on the gas estimator, a deadline/timeout is attached to it by the estimator
+	ctx := context.Background()
```

Comment:
Is there anything to connect this too if we go up the stack to the caller of `estimateGasPrice()`?